crates/db-access/src/queries.rs

```diff
@@ -190,3 +190,58 @@ pub async fn get_block_headers_by_time_range(
 
     Ok(headers)
 }
+
+pub async fn create_job_request(
+    pool: &PgPool,
+    job_id: &str,
+    status: JobStatus,
+) -> Result<(), sqlx::Error> {
+    sqlx::query!(
+        r#"
+        INSERT INTO job_requests (job_id, status) VALUES ($1, $2)
+        "#,
+        job_id,
+        status.as_str()
+    )
+    .execute(pool)
+    .await?;
+
+    Ok(())
+}
+
+pub async fn get_job_request(
+    pool: &PgPool,
+    job_id: &str,
+) -> Result<Option<JobRequest>, sqlx::Error> {
+    sqlx::query_as!(
+        JobRequest,
+        r#"
+        SELECT job_id, status as "status: JobStatus"
```

Comment:
Does this work? Seems a bit weird to me. I did see a post describing this as a solution.